AppController/djinn.rb

```diff
@@ -2283,7 +2285,7 @@ def receive_server_message(timeout, secret)
         # Client will process "Error" and try again unless appscale is
         # terminated
         if @done_terminating and @waiting_messages.empty?
-          return "Error: Done Terminating and No Messages"
+          raise "Error: Done Terminating and No Messages"
```

Comment:
Raising an exception here prevents the server from returning a proper error message. It basically causes the thread to crash and the server to return a 500.
If this changes was motivated by wanting the tools' acc object to raise an exception, you should parse the response there and have the client raise the exception.